### PR TITLE
Bug fixes related to GrayLongwaveRadiation

### PR DESCRIPTION
**Pull request by JoyMonteiro**
Fixed some labelling and init issues related to Gray radiation.

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -6,6 +6,14 @@ Latest
 ------
 
 * Fix docs to reflect API changes after refactor.
+* Fix wrong initialisation of optical depth for GrayLongwaveRadiation
```
Comment (mcgibbon) [inline HISTORY.rst]: Wrong in what way? Can you expand this to be more specific?

File: HISTORY.rst [hunk 2]
```diff
@@ -6,6 +6,14 @@ Latest
...
+
+Breaking Changes
+----------------
+
+* The flux outputs of GrayLongwaveRadiation have been renamed to eliminate
+  `on_interface_levels`
```
Comment (mcgibbon) [inline HISTORY.rst]: add "to be consistent with RRTMGLongwaveRadiation" or something similar.

File: climt/_core/initialization.py [hunk 3]
```diff
@@ -151,9 +151,14 @@ class PressureFunctionDiagnosticComponent(DiagnosticComponent):
 
     input_properties = {
         'air_pressure': {
-            'dims': ['*'],
```
Comment (mcgibbon) [inline climt/_core/initialization.py]: The dims of air pressure and surface air pressure should be different. You probably want to change this to `['mid_levels', '*']`.

File: climt/_core/initialization.py [hunk 4]
```diff
@@ -194,7 +206,7 @@ def __init__(
                 "Argument mid_or_interface_levels must be 'mid' or 'interface'")
         self.diagnostic_properties = {
             output_name: {
-                'dims': ['*'],
+                'dims': [vertical_dimension],
```
Comment (mcgibbon) [inline climt/_core/initialization.py]: This should be the same dimensionality as air_pressure. In other words, `['mid_levels', '*']`.

File: climt/_core/initialization.py [hunk 5]
```diff
@@ -787,7 +799,7 @@ def init_ozone(p):
 init_diagnostics = [
     PressureFunctionDiagnosticComponent(
         'longwave_optical_depth',
-        lambda p: 1. * (1. - p),
+        lambda p, ps: 1. * (1. - p/ps),
```
Comment (mcgibbon) [inline climt/_core/initialization.py]: After making the above changes to dimensionality, use `ps[None, :]`.

Comment (JoyMonteiro) [reply climt/_core/initialization.py]: I did not need to change the dimensionality, seems to work ok!

File: HISTORY.rst [hunk 6]
```diff
@@ -6,13 +6,14 @@ Latest
 ------
 
 * Fix docs to reflect API changes after refactor.
-* Fix wrong initialisation of optical depth for GrayLongwaveRadiation
+* Fix wrong initialisation to use sigma values instead of pressure values
```
Comment (mcgibbon) [inline HISTORY.rst]: initialization, if we're using American spellings.

File: climt/_components/dry_convection/component.py [hunk 7]
```diff
@@ -84,10 +84,11 @@ def array_call(self, state, time_step):
                 convect_to_level = len(theta_lesser) - np.argmax(theta_lesser[::-1])
 
                 if level == 0:
-                    convect_to_level = 1
+                    convect_to_level = max(convect_to_level, 1)
 
                 if convect_to_level == 0:
                     continue
+                print(convect_to_level)
```
Comment (mcgibbon) [inline climt/_components/dry_convection/component.py]: Remove this print statement.

File: tests/test_components.py [hunk 8]
```diff
@@ -452,22 +452,25 @@ def heat_capacity(q):
 
 def test_enthalpy_and_water_conservation():
```
Comment (mcgibbon) [inline tests/test_components.py]: Add a short docstring describing what this is testing. It's not clear to me from the name. Maybe include the name of the component being tested in the function name.

File: HISTORY.rst (hunk 6 above)
Comment (JoyMonteiro) [reply HISTORY.rst]: I will retain British spellings, it is hard for me to write American versions without consciously
remembering to do so.

Comment (mcgibbon) [reply HISTORY.rst]: OK for documentation, though I will insist on American spellings for function/variable names where it's more likely to cause confusion for users.

Comment (JoyMonteiro) [reply HISTORY.rst]: Yes, I'm fine with that!

**Pull request merged**